src/plugins/chart_expressions/expression_partition_vis/public/utils/layers/get_layers.ts

```diff
@@ -53,6 +54,9 @@ export const getLayers = (
   }
 
   const sortPredicateForType = sortPredicateByType(chartType, visParams, visData, columns);
+
+  const distinctSeries = getDistinctSeries(rows, columns);
```

Comment:
This was moved outside the `getColor` function to avoid calling this on every slice of the partition every time we need that color.